airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/RouterService.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling;
+
+import io.airbyte.commons.temporal.scheduling.GeographyMapper;
+import io.airbyte.config.Geography;
+import io.airbyte.config.persistence.ConfigRepository;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Decides which Task Queue should be used for a given connection's sync operations, based on the
+ * configured {@link Geography}
+ */
+@Singleton
+@Slf4j
+public class RouterService {
+
+  private final ConfigRepository configRepository;
+  private final GeographyMapper geographyMapper;
+
+  public RouterService(final ConfigRepository configRepository, final GeographyMapper geographyMapper) {
```

Comment:
This is where Micronaut will inject the Cloud specific bean?

Reply:
Yes.  It will inject which ever implementation of the `GeographyMapper` wins.  In OSS, this will be the default implementation (the only one present).  In cloud, both are present on the classpath, but only the one provided by the `cloud-worker` library will win because of the `Primary` annotation.